httpbin/helpers.go

```diff
@@ -50,6 +50,11 @@ func getClientIP(r *http.Request) string {
 		return clientIP
 	}
 
+	// Special case Cloudflare for the value directly.
+	if cfClientIP := r.Header.Get("CF-Connecting-IP"); cfClientIP != "" {
+		return cfClientIP
```

Comment:
Style nit:  please follow the pattern in the previous conditional and just call this `clientIP` rather than `cfClientIP`.

In case it's not obvious, in Golang's

```go
if x := y; test {}
```

declaration+conditional form, the declared variable does not escape the scope of the conditional, which means we don't need to worry about either shadowing the previous conditional's `clientIP` value or leaking this `clientIP` value into the rest of the function.